density/db/db.py

```diff
@@ -189,7 +189,7 @@ def get_oauth_code_for_uni(cursor, uni):
     else:
         # If the code DNE, create a new one and insert into the database.
         new_code = ''.join(random.choice(
-            string.ascii_uppercase + string.digits) for x in xrange(32))
+            string.ascii_uppercase + string.digits) for _x in xrange(32))
```

Comment:
So the two options I see are:
- base64 encode `os.urandom` output (either concat it to a `uuid` to guarantee uniqueness or check for uniqueness when inserting to the database).
- Encrypt (probably via https://cryptography.io/en/latest/fernet/) a uuid.

I think the first one's a better choice, but your call.